Add raster:bands and spatial resolution summaries

denarch22 opened the pull request:
## 🛠️ Changes
This update annotates each asset with raster:bands metadata using the PySTAC RasterExtension, mapping dtype and spatial resolution from item metadata. It also adds a collection-level summary of available spatial resolutions under 'raster:spatial_resolution'.

## 📝 Associated issues
Resolve [LIB-310](https://linear.app/biodev/issue/LIB-310/agregar-propiedad-de-resolucion-en-rasters)

denarch22 committed:
Add raster:bands and spatial resolution summaries

This update annotates each asset with raster:bands metadata using the PySTAC RasterExtension, mapping dtype and spatial resolution from item metadata. It also adds a collection-level summary of available spatial resolutions under 'raster:spatial_resolution'.

denarch22 committed:
Refactor dtype mapping and formatting in collection.py

Improved readability of the _map_dtype_to_pystac function by expanding its formatting and docstring. Reformatted several code blocks for better clarity and consistency, including set comprehensions, asset creation, and exception logging.

cazapatamar commented on src/collection.py:
```diff
@@ -10,12 +10,35 @@
 from utils.logging_config import logger
 
 
+def _map_dtype_to_pystac(dtype_str: str) -> DataType | None:
+    """Map the detected dtype (e.g. 'uint16', 'float32') to DataType from PySTAC.
+    Expand according to the possible dtypes that raster.get_tif_metadata returns.
+    """
+    d = (dtype_str or "").lower()
+    if d in ("uint8", "ubyte"):
+        return DataType.UINT8
+    if d in ("uint16",):
+        return DataType.UINT16
+    if d in ("uint32",):
+        return DataType.UINT32
+    if d in ("int16",):
+        return DataType.INT16
+    if d in ("int32",):
+        return DataType.INT32
+    if d in ("float32", "float"):
+        return DataType.FLOAT32
+    if d in ("float64", "double"):
+        return DataType.FLOAT64
+    return None
```
@denarch22 revisa si esta funcion puede estar mas bien en `utils`. Me suena que es algo que podemos reusar mas adelante.

Tambien sugiero cambiarle el nombre a la funcion por esta `map_dtype_to_pystac_datatype`

denarch22 committed:
Refactor dtype mapping to stac_helpers module

Moved the raster dtype to PySTAC DataType mapping logic from collection.py to a new utils/stac_helpers.py module for better code organization and reuse.
